### PR TITLE
feat(frontend): gms - new menu item userplayground

### PR DESCRIPTION
**Pull request by clauspeterhuebner**
### Issues

- None

### Todo

- [X] None

### CODE REVIEW

File: frontend/public/img/svg/loupe.png
Comment (einhornimmond) [inline frontend/public/img/svg/loupe.png:1]: a png file in a svg folder?
Please move to frontend/public/img/ or make an actual svg out of it.

File: backend/.env.template [hunk 1]
```diff
@@ -66,4 +66,7 @@ FEDERATION_XCOM_SENDCOINS_ENABLED=$FEDERATION_XCOM_SENDCOINS_ENABLED
 
 # GMS
 GMS_ACTIVE=$GMS_ACTIVE
-GMS_URL=$GMS_URL
+GMS_API_URL=$GMS_API_URL
+GMS_DASHBOARD_URL=$GMS_DASHBOARD_URL
+GMS_WEBHOOK_SECRET=$GMS_WEBHOOK_SECRET
+GMS_CREATE_USER_THROW_ERRORS=$GMS_CREATE_USER_THROW_ERRORS
```
Comment (einhornimmond) [inline backend/.env.template:72]: please add to deployment/bare_metal/.env.dist

File: backend/src/apis/gms/GmsClient.ts [hunk 2]
```diff
@@ -118,7 +118,9 @@ export async function userByUuid(uuid: string): Promise<GmsUser[] | string | und
 
 export async function createGmsUser(apiKey: string, user: GmsUser): Promise<boolean> {
   if (CONFIG.GMS_ACTIVE) {
-    const baseUrl = CONFIG.GMS_URL.endsWith('/') ? CONFIG.GMS_URL : CONFIG.GMS_URL.concat('/')
+    const baseUrl = CONFIG.GMS_API_URL.endsWith('/')
+      ? CONFIG.GMS_API_URL
+      : CONFIG.GMS_API_URL.concat('/')
```
Comment (einhornimmond) [inline backend/src/apis/gms/GmsClient.ts:123]: https://de.wikibrief.org/wiki/Rule_of_three_%28computer_programming%29

File: backend/src/apis/gms/GmsClient.ts [hunk 3]
```diff
@@ -152,7 +154,9 @@ export async function createGmsUser(apiKey: string, user: GmsUser): Promise<bool
 
 export async function updateGmsUser(apiKey: string, user: GmsUser): Promise<boolean> {
   if (CONFIG.GMS_ACTIVE) {
-    const baseUrl = CONFIG.GMS_URL.endsWith('/') ? CONFIG.GMS_URL : CONFIG.GMS_URL.concat('/')
+    const baseUrl = CONFIG.GMS_API_URL.endsWith('/')
+      ? CONFIG.GMS_API_URL
+      : CONFIG.GMS_API_URL.concat('/')
```
Comment (einhornimmond) [inline backend/src/apis/gms/GmsClient.ts:159]: https://de.wikibrief.org/wiki/Rule_of_three_%28computer_programming%29

File: backend/src/apis/gms/GmsClient.ts [hunk 4]
```diff
@@ -189,7 +193,9 @@ export async function verifyAuthToken(
   communityUuid: string,
   token: string,
 ): Promise<string> {
-  const baseUrl = CONFIG.GMS_URL.endsWith('/') ? CONFIG.GMS_URL : CONFIG.GMS_URL.concat('/')
+  const baseUrl = CONFIG.GMS_API_URL.endsWith('/')
+    ? CONFIG.GMS_API_URL
+    : CONFIG.GMS_API_URL.concat('/')
   const service = 'verify-auth-token?token='.concat(token).concat('&uuid=').concat(communityUuid)
```
Comment (einhornimmond) [inline backend/src/apis/gms/GmsClient.ts:199]: https://de.wikibrief.org/wiki/Rule_of_three_%28computer_programming%29

File: frontend/src/components/Menu/Sidebar.vue [hunk 5]
```diff
@@ -28,10 +28,14 @@
...
           </b-nav-item>
-          <b-nav-item to="/information" active-class="activeRoute">
+          <b-nav-item to="/information" class="mb-3" active-class="activeRoute">
             <b-img src="/img/svg/info.svg" height="20" class="svg-icon" />
             <span class="ml-2">{{ $t('navigation.info') }}</span>
           </b-nav-item>
+          <b-nav-item to="/usersearch" active-class="activeRoute">
+            <b-img src="/img/svg/loupe.png" height="20" class="svg-icon" />
+            <span class="ml-2">{{ $t('navigation.usersearch') }}</span>
+          </b-nav-item>
```
Comment (einhornimmond) [inline frontend/src/components/Menu/Sidebar.vue:38]: it is a png icon, so please don't use svg-icon class, maybe this is the cause for the black symbol, if choosen.

**Review by einhornimmond — CHANGES_REQUESTED** [review]
please fix at least the black box symbol after choosing nutzersuche.
grafik

File: frontend/public/img/svg/loupe.png
Comment (clauspeterhuebner) [reply frontend/public/img/svg/loupe.png:1]: done

File: backend/.env.template (hunk 1 above)
Comment (clauspeterhuebner) [reply backend/.env.template:72]: done

File: backend/src/apis/gms/GmsClient.ts (hunk 2 above)
Comment (clauspeterhuebner) [reply backend/src/apis/gms/GmsClient.ts:123]: done

File: backend/src/apis/gms/GmsClient.ts (hunk 3 above)
Comment (clauspeterhuebner) [reply backend/src/apis/gms/GmsClient.ts:159]: done

File: backend/src/apis/gms/GmsClient.ts (hunk 4 above)
Comment (clauspeterhuebner) [reply backend/src/apis/gms/GmsClient.ts:199]: done

File: frontend/src/components/Menu/Sidebar.vue (hunk 5 above)
Comment (clauspeterhuebner) [reply frontend/src/components/Menu/Sidebar.vue]: Danke für deine wertvollen Anmerkungen...

File: backend/src/graphql/resolver/util/authenticateGmsUserPlayground.ts [hunk 6]
```diff
@@ -10,7 +10,11 @@ export async function authenticateGmsUserPlayground(
   dbUser: DbUser,
 ): Promise<GmsUserAuthenticationResult> {
   const result = new GmsUserAuthenticationResult()
-  result.url = CONFIG.GMS_URL.concat('/playground')
+  const dashboardUrl = CONFIG.GMS_DASHBOARD_URL.endsWith('/')
+    ? CONFIG.GMS_DASHBOARD_URL
+    : CONFIG.GMS_DASHBOARD_URL.concat('/')
+
```
Comment (einhornimmond) [inline backend/src/graphql/resolver/util/authenticateGmsUserPlayground.ts:16]: I like ensureUrlEndsWithSlash, maybe you can it use here also?

**Review by einhornimmond — APPROVED** [review]

File: backend/src/graphql/resolver/util/authenticateGmsUserPlayground.ts (hunk 6 above)
Comment (clauspeterhuebner) [reply backend/src/graphql/resolver/util/authenticateGmsUserPlayground.ts:16]: resolved this and in several files of backend too...

Comment (clauspeterhuebner) [reply backend/src/graphql/resolver/util/authenticateGmsUserPlayground.ts:16]: resolved this and in several files of backend too...

**Pull request merged**